staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go

```diff
@@ -0,0 +1,42 @@
+package metrics
+
+import (
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	etcdRequestLatencies = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "etcd_request_latency_seconds",
+			Help:    "Etcd request latency in seconds for each operation and object type.",
+			Buckets: prometheus.ExponentialBuckets(0.001, 5, 7),
+		},
+		[]string{"operation", "type"},
+	)
+)
+
+var registerMetrics sync.Once
+
+// Register all metrics.
+func Register() {
+	// Register the metrics.
+	registerMetrics.Do(func() {
+		prometheus.MustRegister(etcdRequestLatencies)
+	})
+}
+
+func RecordEtcdRequestLatency(verb, resource string, startTime time.Time) {
```

Comment:
remove Etcd in the method naming. this will be used as etcdmetrics.RecordRequestLatency